lib/metadataTypes/Script.js

```diff
@@ -53,15 +53,17 @@ class Script extends MetadataType {
      * @param {string} templateDir Directory where retrieved metadata directory will be saved
      * @param {string} name name of the metadata file
      * @param {Util.TemplateMap} templateVariables variables to be replaced in the metadata
-     * @returns {Promise<{metadata:ScriptMap,type:string}>} Promise
+     * @returns {Promise<{metadata:Script,type:string}>} Promise
      */
     static async retrieveAsTemplate(templateDir, name, templateVariables) {
         await File.initPrettier('ssjs');
-        return super.retrieveREST(
-            templateDir,
-            '/automation/v1/scripts/?$filter=name%20eq%20' + encodeURIComponent(name),
-            null,
-            templateVariables
+        return Util.reduceMetadataMap(
+            await super.retrieveREST(
```

Comment:
open point for discussion if retrieveREST wouldnt be the better place to put this? we could conditionally change result based on template.

Reply:
Good idea. Let me check the feasibility when I get back to my laptop (travelling atm)

Reply:
done